### PR TITLE
Feat(eos_cli_config_gen): Add MPLS Tunnel Support for Traceroute and PMTU Discovery

### LINKED ISSUE #4185: MPLS Tunnel Support for Traceroute and PMTU Discovery
### Enhancement summary

The following TOI describes support for traceroute in MPLS networks and PMTU discovery:
https://www.arista.com/en/support/toi/eos-4-21-0f/14043-mpls-tunnel-support-for-traceroute-and-pmtu-discovery

The following two commands are required which are not supported today in AVD natively:
```
  mpls icmp fragmentation-needed tunneling
  mpls icmp ttl-exceeded tunneling 
```

### Which component of AVD is impacted

eos_cli_config_gen

### Use case example

MPLS IPVPNv4 network to support traceroutes

### Describe the solution you would like

Support for this two commands via eos_cli_config_gen

### Describe alternatives you have considered

eos_cli: |
  mpls icmp fragmentation-needed tunneling
  mpls icmp ttl-exceeded tunneling 

### Additional context

_No response_

### Contributing Guide

- [X] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by Shivani-gslab**
## Change Summary

Add MPLS Tunnel Support for Traceroute and PMTU Discovery

## Related Issue(s)

Fixes #4185

## Component(s) name

`arista.avd.eos_cli_config_gen`

## Proposed changes
Implementing schema and template for - 
```
mpls icmp fragmentation-needed tunneling
mpls icmp ttl-exceeded tunneling 
```

## How to test
Test in EOS CLI and the molecule tests in this PR.

### Self review Checklist

- [x] Order of config should be as per EOS cli.
- [x] Add exclamatory marks(!) at correct position in templates. It should be same as reflected in run-config.
- [x] Verify the changes by running pre-commit and all molecules.
- [x] Along with EOS config template, update the documentation template as well (if required).
- [x] Add the test-case in molecules for any new data-model/key/fix.
- [x] Add description to the keys as per the docs.
- [x] Add punctuation to all descriptions.

### Repository Checklist

- [x] My code has been rebased from devel before I start
- [x] I have read the [**CONTRIBUTING**](https://avd.sh/en/latest/docs/contribution/overview.html) document.
- [x] My change requires a change to the documentation and documentation have been updated accordingly.
- [x] I have updated [molecule CI](https://github.com/aristanetworks/avd/tree/devel/ansible_collections/arista/avd/molecule) testing accordingly. (check the box if not applicable)

### CODE REVIEW

**Comment by laxmikantchintakindi** [comment]
LGTM

File: ansible_collections/arista/avd/molecule/eos_cli_config_gen/intended/configs/mpls.cfg [hunk 1]
```diff
@@ -30,5 +30,8 @@ mpls ldp
    router-id 192.168.1.1
    no shutdown
    transport-address interface Loopback0
+!
+   mpls icmp fragmentation-needed tunneling
+   mpls icmp ttl-exceeded tunneling
```
Comment (gmuloc) [inline ansible_collections/arista/avd/molecule/eos_cli_config_gen/intended/configs/mpls.cfg]: wrong indentation
```suggestion
mpls icmp fragmentation-needed tunneling
mpls icmp ttl-exceeded tunneling
```

File: python-avd/pyavd/_eos_cli_config_gen/j2templates/eos/mpls.j2 [hunk 2]
```diff
@@ -26,3 +26,12 @@ mpls ldp
...
 {%     endif %}
 {% endif %}
+{% if mpls.icmp is arista.avd.defined %}
+!
+{%     if mpls.icmp.fragmentation_needed_tunneling is arista.avd.defined(true) %}
+   mpls icmp fragmentation-needed tunneling
+{%     endif %}
+{%     if mpls.icmp.ttl_exceeded_tunneling is arista.avd.defined(true) %}
+   mpls icmp ttl-exceeded tunneling
+{%     endif %}
```
Comment (gmuloc) [inline python-avd/pyavd/_eos_cli_config_gen/j2templates/eos/mpls.j2]:
```suggestion
mpls icmp fragmentation-needed tunneling
{%     endif %}
{%     if mpls.icmp.ttl_exceeded_tunneling is arista.avd.defined(true) %}
mpls icmp ttl-exceeded tunneling
{%     endif %}
```

**Comment by gmuloc** [comment]
@dgonzalez85 can you please confirm thats what is needed. Also I found indication that
Command `mpls icmp fragmentation-needed tunneling` is not implemented in all platforms. Can you please confirm as it probably should go in the description

**Review by gmuloc — CHANGES_REQUESTED** [review]

**Comment by dgonzalez-arista** [comment]
> m thats what is needed. Also I found indication th

Looks good! I will test. As we've discussed, MPLS support was mainly in Jericho-based platforms, but seems there are some basic in strata as well: https://www.arista.com/en/support/toi/eos-4-23-0f/14327-mpls-support-on-x-series-switches. So good to add a note to review platform support. thanks!

**Comment by gmuloc** [comment]
> > m thats what is needed. Also I found indication th
> 
> Looks good! I will test. As we've discussed, MPLS support was mainly in Jericho-based platforms, but seems there are some basic in strata as well: https://www.arista.com/en/support/toi/eos-4-23-0f/14327-mpls-support-on-x-series-switches. So good to add a note to review platform support. thanks!

Thanks @dgonzalez85 - @Shivani-gslab after some thought we are not putting everywhere "this is platform dependent" (e.g. for the command only supported on CloudEOS) so I would leave it as it is.

File: python-avd/tox.ini [hunk 3]
```diff
@@ -55,3 +55,4 @@ deps =
     coverage[toml]
 commands =
     coverage report --rcfile=pyproject.toml
+    coverage html --rcfile=pyproject.toml
```
Comment (Shivani-gslab) [inline python-avd/tox.ini]:
```suggestion  
```

File: python-avd/pyavd/_eos_cli_config_gen/j2templates/documentation/mpls-and-ldp.j2 [hunk 4]
```diff
@@ -17,6 +17,8 @@
 | LDP Router ID | {{ mpls.ldp.router_id | arista.avd.default('-') }} |
 | LDP Interface Disabled Default | {{ mpls.ldp.interface_disabled_default | arista.avd.default('-') }} |
 | LDP Transport-Address Interface | {{ mpls.ldp.transport_address_interface | arista.avd.default('-') }} |
+| ICMP Fragmentation-Needed Tunneling Enabled | {{ mpls.icmp.fragmentation_needed_tunneling | arista.avd.default('-') }} |
+| ICMP TTL-Exceeded Tunneling Enabled | {{ mpls.icmp.ttl_exceeded_tunneling | arista.avd.default('-') }} |
```
Comment (ClausHolbechArista) [inline python-avd/pyavd/_eos_cli_config_gen/j2templates/documentation/mpls-and-ldp.j2]:
```suggestion
{%     if mpls.icmp.fragmentation_needed_tunneling is arista.avd.defined %}
| ICMP Fragmentation-Needed Tunneling Enabled | {{ mpls.icmp.fragmentation_needed_tunneling }} |
{%     endif %}
{%     if mpls.icmp.ttl_exceeded_tunneling is arista.avd.defined %}
| ICMP TTL-Exceeded Tunneling Enabled | {{ mpls.icmp.ttl_exceeded_tunneling }} |
{%     endif %}
```

**Review by ClausHolbechArista — CHANGES_REQUESTED** [review]

**Review by ClausHolbechArista — APPROVED** [review]

**Review by MaheshGSLAB — APPROVED** [review]

**Review by gmuloc — APPROVED** [review]

**Pull request merged**